lib/Server.js

```diff
@@ -538,6 +538,110 @@ class Server {
     });
   }
 
+  runBonjour(options) {
+    const bonjour = require('bonjour')();
+    const os = require('os');
+    const { https, port } = options;
+
+    bonjour.publish({
+      name: `Webpack Dev Server ${os.hostname()}:${port}`,
+      port,
+      type: https ? 'https' : 'http',
+      subtypes: ['webpack'],
+      ...options.bonjour,
+    });
+
+    process.on('exit', () => {
+      bonjour.unpublishAll(() => {
+        bonjour.destroy();
+      });
+    });
+  }
+
+  openBrowser(uri, options) {
+    const isAbsoluteUrl = require('is-absolute-url');
+    const open = require('open');
+
+    // https://github.com/webpack/webpack-dev-server/issues/1990
+    const defaultOpenOptions = { wait: false };
+    const openTasks = [];
+
+    const getOpenTask = (item) => {
+      if (typeof item === 'boolean') {
+        return [{ target: uri, options: defaultOpenOptions }];
+      }
+
+      if (typeof item === 'string') {
+        return [{ target: item, options: defaultOpenOptions }];
+      }
+
+      let targets;
+
+      if (item.target) {
+        targets = Array.isArray(item.target) ? item.target : [item.target];
+      } else {
+        targets = [uri];
+      }
+
+      return targets.map((target) => {
+        const openOptions = defaultOpenOptions;
+
+        if (item.app) {
+          if (typeof item.app === 'string') {
+            openOptions.app = { name: item.app };
+          } else {
+            openOptions.app = item.app;
+          }
+        }
+
+        return { target, options: openOptions };
+      });
+    };
+
+    if (Array.isArray(options)) {
+      options.forEach((item) => {
+        openTasks.push(...getOpenTask(item));
+      });
+    } else {
+      openTasks.push(...getOpenTask(options));
+    }
+
+    return Promise.all(
```

Comment:
I think we don't need `return` here, because we don't use returned result :smile:

Reply:
Yes, good catch.